### PR TITLE
Added FastCell and DropOut support to SRNN, Readme Corrected

### PR DESCRIPTION
**Pull request by SachinG007**

### CODE REVIEW

File: pytorch/examples/SRNN/SRNN_Example.py [hunk 1]
```diff
@@ -51,8 +51,21 @@
...
+
+Example OPTIONAL args for FastGRNNCell
+cellArgs = {'gate_non_linearity':"sigmoid",'update_non_linearity':"tanh",
+				'wRank':None, 'uRank':None,'zetaInit':1.0, 'nuInit':-4.0, 
+				'batch_first':False}
+
+'''
+cellArgs = {}
+
+srnn2 = SRNN2(numInput, numClasses, hiddenDim0, hiddenDim1, cellType, dropoutProbability, **cellArgs).to(device)
```
Comment (metastableB) [inline pytorch/examples/SRNN/SRNN_Example.py]: break lines please.

File: pytorch/pytorch_edgeml/graph/rnn.py [hunk 2]
```diff
@@ -1050,29 +1049,42 @@ def forward(self, input, hiddenState=None, cellState=None):
 
 class SRNN2(nn.Module):
 
-    def __init__(self, inputDim, outputDim, hiddenDim0, hiddenDim1, cellType):
+    def __init__(self, inputDim, outputDim, hiddenDim0, hiddenDim1, cellType, dropoutProbability, **cellArgs):
```
Comment (metastableB) [inline pytorch/pytorch_edgeml/graph/rnn.py]: 1. break lines.
2. We should have separate dropout number to each layer. Also we should have a default argument for the dropout for people who don't want to use it.
```
def __init__(self, inputDim, outputDim, hiddenDim0, hiddenDim1, cellType,
                   dropout0=1.0, dropout1=1.0, **cellArgs)
```

Even though having a dropout layer with 1.0 dropout (ie. no dropout) might be functionally same as having no dropout, there might be performance hits because of this additional layer. I guess you could default to `dropout0=None` and then not add the dropout layer if `dropout0` remains none inside `__init__`. 

Also, please `assert 0< dropout0 <= 1.0`

File: pytorch/pytorch_edgeml/graph/rnn.py [hunk 3]
```diff
@@ -1050,29 +1049,42 @@ def forward(self, input, hiddenState=None, cellState=None):
...
         '''
         A 2 Layer Shallow RNN.
 
         inputDim: Input data's feature dimension.
         hiddenDim0: Hidden state dimension of the lower layer RNN cell.
         hiddenDim1: Hidden state dimension of the second layer RNN cell.
-        cellType: The type of RNN cell to use. Options are ['LSTM']
+        cellType: The type of RNN cell to use. Options are ['LSTM', 'FastRNNCell', 'FastGRNNCell', 'GRULRCell']
```
Comment (metastableB) [inline pytorch/pytorch_edgeml/graph/rnn.py]: Please break lines.

File: pytorch/pytorch_edgeml/graph/rnn.py [hunk 4]
```diff
@@ -1050,29 +1049,42 @@ def forward(self, input, hiddenState=None, cellState=None):
...
+        
         if self.cellType == 'LSTM':
             self.rnnClass = nn.LSTM
+        
+        elif self.cellType == 'FastRNNCell':
+            self.rnnClass = FastRNN
+
+        elif self.cellType == 'FastGRNNCell':
+            self.rnnClass = FastGRNN
+
```
Comment (metastableB) [inline pytorch/pytorch_edgeml/graph/rnn.py]: remove unnecessary new lines.

File: pytorch/examples/SRNN/SRNN_Example.py (hunk 1 above)
Comment (SachinG007) [reply pytorch/examples/SRNN/SRNN_Example.py]: Finalizing to using 72 chars per line from now onwards
